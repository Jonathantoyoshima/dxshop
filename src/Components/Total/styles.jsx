import styled from 'styled-components';

export const Root = styled.div`
  background-color: ${(props) => props.theme.color.primary.light};
  display: flex;
  color: ${(props) => props.theme.color.primary.dark};
  font-weight: bold;
  font-size: 1.2rem;
  padding: 16px;
`;

export const Text = styled.div`
  flex: 1;
`;

export const Value = styled.div``;
