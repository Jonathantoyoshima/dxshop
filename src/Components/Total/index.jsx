import { Root, Value, Text } from './styles';

const Total = () => {
  return (
    <Root>
      <Text>Total: </Text>
      <Value>R$ 3,50</Value>
    </Root>
  );
};

const defaultProps = {
  theme: {
    color: {
      basic: {
        black: '#0c0c0c',
        white: '#fff',
      },
      primary: {
        light: '#000',
        main: '#ccc',
        dark: '#000',
      },
    },
  },
};
Text.defaultProps = defaultProps;
Value.defaultProps = defaultProps;
Root.defaultProps = defaultProps;

export default Total;
