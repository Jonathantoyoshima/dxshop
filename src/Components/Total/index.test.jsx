import React from 'react';
import renderer from 'react-test-renderer';
import Total from './index';

test('Render Build Total Snapshot', () => {
  const component = renderer.create(<Total />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
