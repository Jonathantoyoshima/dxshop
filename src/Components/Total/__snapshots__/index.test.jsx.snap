// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Build Total Snapshot 1`] = `
<div
  className="sc-bdfBwQ dWfjBh"
>
  <div
    className="sc-gsTCUz cXavrX"
  >
    Total: 
  </div>
  <div
    className="sc-dlfnbm"
  >
    R$ 3,50
  </div>
</div>
`;
